src/crystal/system/unix/getrandom.cr

```diff
@@ -33,7 +33,7 @@ module Crystal::System::Random
       return unless urandom.info.type.character_device?
 
       urandom.close_on_exec = true
-      urandom.sync = true # don't buffer bytes
+      urandom.read_buffering = false # don't buffer bytes
```

Comment:
For context: `sync` controls write buffering, which is completely irrelevant here. `read_buffering` is used to control read buffering.